azure-pipelines.yml

```diff
@@ -55,6 +55,76 @@ stages:
           - checkout: self
           - script: ./ci/kotlin_lint.sh
             displayName: 'Run Kotlin lint'
+  # Note: the size regression jobs are not ideal.
+  # The size analysis described in the docs was ran on arm64 machines.
+  # However, it seems that azure pipelines will support arm64 machines sometime soon.
+  # Whenever that is available this jobs should be moved to that architecture.
+  # Issue: https://github.com/lyft/envoy-mobile/issues/216
+  - stage: performance
+    dependsOn: [] # this removes the implicit dependency on previous stage and causes this to run in parallel.
+    jobs:
+      - job: size_current
+        timeoutInMinutes: 60
+        pool:
+          vmImage: 'Ubuntu 16.04'
+        steps:
+          - checkout: self
+            submodules: true
+          - script: ./ci/linux_ci_setup.sh
+            displayName: 'Install dependencies'
+          - script: bazel build //test/performance:test_binary_size --config=sizeopt
+            displayName: 'Build test binary'
+          - task: PublishPipelineArtifact@0
+            displayName: 'Publish current test binary'
+            inputs:
+              artifactName: 'current'
+              targetPath: bazel-bin/test/performance/test_binary_size
+      - job: size_master
+        timeoutInMinutes: 60
+        pool:
+          vmImage: 'Ubuntu 16.04'
+        steps:
+          - checkout: self
+            submodules: true
+          - script: git checkout master && git pull origin master
```

Comment:
In the future, we may want to consider:
- Running this against the SHA that the PR was opened against (i.e., if this branch is 5 commits behind master, compare against that SHA instead of the tip of master which may have changed since the PR was opened/rebased)
- Caching these artifacts so we don't have to rebuild master against each PR

Reply:
Good point on caching: https://github.com/lyft/envoy-mobile/issues/229.

I see your point. But why should we prefer the time at which the PR was forked from master than the current tip?

Reply:
I think so, using master here almost feels like a side effect of sorts